### PR TITLE
Strings - search and replace fixes

### PR DESCRIPTION
**Pull request by tethryus**
Treats empty substrings
Updated test cases

### CODE REVIEW

File: content/io/cloudslang/base/strings/search_and_replace.sl [hunk 1]
```diff
@@ -25,11 +27,19 @@ operation:
...
-        replaced_string = origin_string.replace(text_to_replace, replace_with)
-      else:
-        replaced_string = origin_string
+      try:
+        error_message = ""
+        if text_to_replace in origin_string:
+          replaced_string = origin_string.replace(text_to_replace, replace_with)
+        elif text_to_replace not in origin_string:
+          error_message = "Substring not found"
+        else:
```
Comment (mihaitusa) [inline content/io/cloudslang/base/strings/search_and_replace.sl]: @tethryus  So, when you reach the else condition?

Comment (tethryus) [inline content/io/cloudslang/base/strings/search_and_replace.sl]: Fixed it.

File: content/io/cloudslang/base/strings/search_and_replace.sl [hunk 2]
```diff
@@ -12,7 +12,9 @@
 #! @input text_to_replace: text to replace
 #! @input replace_with: text to replace with
 #! @output replaced_string: string with the text replaced
-#! @result SUCCESS: always
+#! @output error_message: substirng not found
```
Comment (mihaitusa) [inline content/io/cloudslang/base/strings/search_and_replace.sl]: typo in "substirng"

File: content/io/cloudslang/base/strings/search_and_replace.sl [hunk 3]
```diff
@@ -25,11 +27,18 @@ operation:
...
-      if text_to_replace in origin_string:
-        replaced_string = origin_string.replace(text_to_replace, replace_with)
-      else:
-        replaced_string = origin_string
+      try:
+        error_message = ""
+        if text_to_replace in origin_string:
+          replaced_string = origin_string.replace(text_to_replace, replace_with)
+        else:
+          if text_to_replace not in origin_string:
```
Comment (mihaitusa) [inline content/io/cloudslang/base/strings/search_and_replace.sl]: @tethryus  You don't need this here right?

**Comment by mihaitusa** [comment]
@tethryus Please handle the comments.

File: content/io/cloudslang/base/strings/search_and_replace.sl (hunk 2 above)
Comment (tethryus) [inline content/io/cloudslang/base/strings/search_and_replace.sl]: Thank you for noticing the typo. Fixed and pushed.

**Comment by tethryus** [comment]
Comments handled, fixes applied.

**Pull request merged**